google/ggrc-core
# GGRC-8533 Remove Review for 5OBJ

# Dependencies

_**This PR should be merged into [GGRC-7200](https://github.com/google/ggrc-core/tree/GGRC-7200) branch which is the common one for all 5OBJ. PRs**_

# Issue description

The support for Review should be removed for all 5OBJ. The already existing Snapshots (and Revisions) should keep the information about review, but the new ones should not have it.

# Steps to test the changes

1. Run the necessary migrations;
2. Make sure that Review could not be created for 5OBJ via REST API for **internal** users.
3. Make sure that Review could not be created for 5OBJ via REST API for **external** users.

# Solution description

This PR contains following changes:

* Remove `Reviewable` mixin from 5OBJ parents.

* Change the creation of `reviewable` field on `Review` class from `json_polymorphic_relationship_factory` to direct usage of `JsonPolymorphicRelationship`. This way there will be no need to specify `Reviewable` as desired type for `Review` object, which allows to perform validation of reviewable object type via corresponding validator provided for `reviewable_type` field.

* Adjust `JsonPolymorphicRelationship.__call__()` to raise `ValueError ` instead of `AssertionError`. `AssertionErrors` exceptions are not handled by our resource class which leads to HTTP responses with 500 status code and this is not correct. `ValidationError` on the other hand is handled by resource class well.

* Provide integration tests that check that no review could be created for 5OBJ via both internal and external API.

# Sanity checklist




# PR Review checklist
